bindsnet/learning/__init__.py

```diff
@@ -374,7 +386,8 @@ def __init__(self, connection: AbstractConnection, nu: Optional[Union[float, Seq
             connection=connection, nu=nu, weight_decay=weight_decay, **kwargs
         )
 
-        assert self.source.traces and self.target.traces, 'Both pre- and post-synaptic nodes must record spike traces.'
+        assert not self.source.traces and not self.target.traces, \
```

Comment:
You can just remove this `assert`. There may be other reasons for recording spike traces.